src/test/api.test.js

```diff
@@ -0,0 +1,156 @@
+});
+
+tap.teardown(async () => {
+  process.exit(0);
+});
+
+tap.test('controller/api', async (t) => {
+  const metadataPath = path.join(__dirname, '/data/metadata');
+  const config = await readConfig(metadataPath);
+
+  t.test('.config()', async (t) => {
+    t.test('when required fields are missing or invalid', async (t) => {
+      t.test('when `rawMetadata` is empty', async (t) => {
+        const body = Object.assign({}, config[0]);
+        delete body['rawMetadata'];
+
+        try {
+          await apiController.config(body);
+          t.fail('Expecting JacksonError.');
+        } catch (err) {
+          t.equal(err.message, 'Please provide rawMetadata');
+          t.equal(err.statusCode, 400);
+        }
+
+        t.end();
+      });
+
+      t.test('when `defaultRedirectUrl` is empty', async (t) => {
+        const body = Object.assign({}, config[0]);
+        delete body['defaultRedirectUrl'];
+
+        try {
+          await apiController.config(body);
+          t.fail('Expecting JacksonError.');
+        } catch (err) {
+          t.equal(err.message, 'Please provide a defaultRedirectUrl');
+          t.equal(err.statusCode, 400);
+        }
+
+        t.end();
+      });
+
+      t.test('when `redirectUrl` is empty', async (t) => {
+        const body = Object.assign({}, config[0]);
+        delete body['redirectUrl'];
+
+        try {
+          await apiController.config(body);
+          t.fail('Expecting JacksonError.');
+        } catch (err) {
+          t.equal(err.message, 'Please provide redirectUrl');
+          t.equal(err.statusCode, 400);
+        }
+
+        t.end();
+      });
+
+      t.test('when `tenant` is empty', async (t) => {
+        const body = Object.assign({}, config[0]);
+        delete body['tenant'];
+
+        try {
+          await apiController.config(body);
+          t.fail('Expecting JacksonError.');
+        } catch (err) {
+          t.equal(err.message, 'Please provide tenant');
+          t.equal(err.statusCode, 400);
+        }
+
+        t.end();
+      });
+
+      t.test('when `product` is empty', async (t) => {
+        const body = Object.assign({}, config[0]);
+        delete body['product'];
+
+        try {
+          await apiController.config(body);
+          t.fail('Expecting JacksonError.');
+        } catch (err) {
+          t.equal(err.message, 'Please provide product');
+          t.equal(err.statusCode, 400);
+        }
+
+        t.end();
+      });
+
+      t.test('when `rawMetadata` is not a valid XML', async (t) => {
+        const body = Object.assign({}, config[0]);
+        body['rawMetadata'] = 'not a valid XML';
+
+        try {
+          await apiController.config(body);
+          t.fail('Expecting JacksonError.');
+        } catch (err) {
+          t.equal(err.statusCode, 400);
+        }
+
+        t.end();
+      });
+    });
+
+    t.test('when the request is good', async (t) => {
+      const body = Object.assign({}, config[0]);
+
+      sinon
```

Comment:
Just thinking out loud here. Curious to get your feedback too :)
- Should we add an assertion to check if keyDigest is called ? 
- clientID could be a [snapshot](https://node-tap.org/docs/api/snapshot-testing/#testing-with-snapshots) (Maybe an overkill)
- possibly an assertion to check if `configStore.put` is also called so that we can verify config is being saved?

Reply:
Thanks for your comment. 

- Yes, it would be good to assert `calledOnce` for `keyDigest`
- I will check the snapshot. Maybe add it later.
- Assertion to check if `configStore.put` need another mocking. This is a private method within the module. Any thoughts?

Reply:
Indeed mocking `configStore` seems a little tricky right now. Is it possible to require just the `apiController` (instead of index.js) and pass our own mocked `configStore` in our test file ?

Reply:
@devkiran @niwsa Instead of mocking configStore we could use the get config method to verify the config has been stored?

Reply:
Yes, so that will cover the GET method also. I can add that along with #35  to the test once this PR is merged.